src/app/api/tasks/tasks.service.ts

```diff
@@ -87,6 +89,12 @@ export class TasksService extends BaseService {
       include: { workflowState: true },
     })
 
+    if (newTask) {
+      // activityEvents.emit('post', newTask.id, this.user)
+      const activityLog = new ActivityLogger({ taskId: newTask.id, user: this.user })
+      await activityLog.createTaskLog()
```

Comment:
it would be great if we can just pass the payload based on the type

```
await activityLogger.log(LogType.TASK_CREATED, payload);
```
and logger will validate based on the type
that way it becomes resuable service

Reply:
Changed in 399e102039aafe762671d041ae19a51ff7c04a83 commit